src/common/lib/transport/connectionmanager.ts

```diff
@@ -364,8 +364,20 @@ class ConnectionManager extends EventEmitter {
             'reattempting connection'
           );
           this.requestState({ state: 'connecting' });
+        } else if (this.state == this.states.connecting) {
+          // RTN20c: if 'online' event recieved while CONNECTING, abandon connection attempt and retry
+          this.pendingTransports.forEach(function (transport) {
+            // Detach transport listeners to avoid connection state side effects from calling dispose
+            transport.off();
+
+            transport.dispose();
+          });
+          this.pendingTransports = [];
+
+          this.startConnect();
```

Comment:
(was about to complain that if this happens near the end of the connecting sequence it might not have enough time to actually connect, but I've just noticed that startConnect() does reset the transition timer so that's fine)